File: CrmApp.Client/src/models/Dictionaries.ts
import { DictionariesElements } from "./DictionariesElements";
import { EDictionaryType } from "./enums/EDictionaryType";
import { UsersProfiles } from "./UsersProfiles";

export class Dictionaries {
    id: number = 0;
    name: string = "";
    description?: string = "";
    dictionaryType: EDictionaryType = EDictionaryType.Custom;
    createdAt: Date = new Date();
    createdBy?: number;
    isCustom: boolean = true;
    isActive: boolean = true;
    createdByUser?: UsersProfiles;
    dictionariesElements: DictionariesElements[] = [];
}

File: CrmApp.Domain/Entities/Dictionaries.cs
using CrmApp.Domain.DTO;
using CrmApp.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CrmApp.Domain.Entities
{
    public class Dictionaries
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; } = string.Empty;
        public EDictionaryType DictionaryType { get; set; }
        public DateTime CreatedAt { get; set; }
        public int? CreatedBy { get; set; }
        public bool IsCustom { get; set; }
        public bool IsActive { get; set; }

        [ForeignKey(nameof(CreatedBy))]
        public virtual UsersProfiles? CreatedByUser { get; set; }
        [NotMapped]
        public List<DictionariesElementsDTO> DictionariesElements { get; set; } = [];
    }
}
